src/Shared/ModelBinding/FormModelBindingHelper.cs

```diff
@@ -0,0 +1,18 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding;
+
+internal static class FormModelBindingHelper
+{
+    public const string CultureInvariantFieldName = "__Invariant";
+
+    public static bool InputTypeUsesCultureInvariantFormatting(string? inputType)
+        => inputType switch
+        {
+            "number" => true,
```

Comment:
There are other inputs requiring invariant formatting, still need to add them here. In fact, I'm realizing as I'm writing this that it's probably the _majority_ of `<input />` types that would use invariant formatting. If this is the case, we may want to invert this and instead emit an extra form field for each occurrence of a culture-specific entry.

Reply:
Update: I've just added all of the input types that use invariant formatting. Maybe we could scrutinize this list and eliminate input types where either culture-specific or culture-invariant parsing/formatting works (like in the checkbox case). That makes this change affect as few inputs as possible. We would just need to make sure we're not put in a situation where we need to modify this list later, since that would be another breaking change.